code/datums/chatmessage.dm

```diff
@@ -125,6 +125,13 @@
 		target.chat_color_darkened = colorize_string(target.name, 0.85, 0.85)
 		target.chat_color_name = target.name
 
+	// Skyrat add
+	var/chat_color_player = owned_by.prefs.read_preference(/datum/preference/color/chat_color_player)
+	if(chat_color_player != "#000000")
+		target.chat_color = sanitize_hexcolor(chat_color_player)
+		target.chat_color_darkened = sanitize_hexcolor(chat_color_player)
```

Comment:
Anyone got any clever string tricks to make the `chat_colored_darkened` a darker tone of the `chat_color_player`'s choice?